controllers/library.py

```diff
@@ -0,0 +1,112 @@
+            new_library_item_document = item_pydantic_to_mongoengine(new_library_item)
+            insert_document_dal('library-items', new_library_item_document)
+            logging.info(
+                f"The library item {new_library_item.title} with ISBN {new_library_item.isbn} was added to '{self.name}' library"
+            )
+
+    def add_new_patron_to_the_library(self, patrons_to_add: List[Patron]):
+        """
+        Add new patrons to the library system.
+
+        Args:
+            patrons_to_add (List[Patron]): List of patrons to add.
+        """
+        for patron in patrons_to_add:
+            if patron.patron_id in self.patrons:
+                raise NotUniquePatronError(f"The patron with ID {patron.patron_id} already exists in the library")
+            self.patrons[patron.patron_id] = patron
+            patron_document = patron_pydantic_to_mongoengine(patron)
+            insert_document_dal('library-patrons', patron_document)
+            logging.info(f"The patron {patron.name} with ID {patron.patron_id} was added to the library")
+
+    def remove_patrons_from_the_library(self, patrons_to_remove: List[Patron]):
+        """
+        Remove patrons from the library system.
+
+        Args:
+            patrons_to_remove (List[Patron]): List of patrons to remove.
+        """
+
+        for patron in patrons_to_remove:
+            if patron.patron_id not in self.patrons:
+                raise NotExistingPatron(f"The patron with ID {patron.patron_id} does not exist in the library")
+            del self.patrons[patron.patron_id]
+            delete_document_dal('library-patrons', {'patron_id': patron.patron_id})
+            logging.info(f"The patron {patron.patron_id} was removed from the library")
+
+    def search_library_items(self, library_item_title=None, library_item_isbn=None):
+        """
+        Search for library items by title or ISBN.
+
+        Args:
+            library_item_title (str): Title to filter by.
+            library_item_isbn (str): ISBN to filter by.
+
+        Returns:
+            List[str]: List of matching item titles.
+        """
+        results = []
+        for library_item in self.library_items.values():
+            if (library_item_title is None or library_item_title in library_item.title) or \
+                    (library_item_isbn is None or library_item_isbn == library_item.isbn):
+                results.append(library_item.title)
+        logging.info(f"Filter matches: {results}")
+        return results
+
+    def remove_library_item_from_the_library(self, library_item_isbn: str):
+        """
+        Remove an item from the library system by ISBN.
+
+        Args:
+            library_item_isbn (str): ISBN of the item to remove.
+        """
+        if library_item_isbn not in self.library_items.keys():
+            raise DoesNotExist(f"The library item with ISBN {library_item_isbn} does not exist in the library")
```

Comment:
why do you have a `NotExistingPatron` error but not the same error for library items?